WooCommerce/Classes/ViewRelated/Themes/ThemesPreviewView.swift

```diff
@@ -126,6 +140,56 @@ struct ThemesPreviewView: View {
             }
             .navigationBarTitleDisplayMode(.inline)
         }
+        .task { await viewModel.fetchPages() }
+        .sheet(isPresented: $showPagesMenu) {
+            pagesListSheet(pages: viewModel.pages)
+        }
+    }
+
+    private func pageSelector(state: ThemesPreviewViewModel.State, pageTitle: String) -> some View {
+        switch state {
+        case .pagesLoading:
+            return AnyView(Text(Localization.preview))
+
+        case .pagesContent:
+            return AnyView(
+                Button(action: { showPagesMenu = true },
+                       label: {
+                           HStack {
+                               Text(pageTitle).bodyStyle()
+                               Image(uiImage: .chevronDownImage)
+                                   .fixedSize()
+                                   .secondaryBodyStyle()
+
+                           }
+                       })
+            )
+
+        case .pagesLoadingError:
+            return AnyView(Text(Localization.preview))
+        }
+    }
```

Comment:
Please rewrite this using `ViewBuilder` to avoid `AnyView` usage. [Why?](https://www.swiftbysundell.com/articles/avoiding-anyview-in-swiftui/) You can directly access the viewModel instead of passing parameters.

Reply:
Great tips, thanks! Added in https://github.com/woocommerce/woocommerce-ios/pull/11492/commits/98fd07bf35f6c4bf9e9dfb75895a462fcd5b47c6